sortinghat/core/api.py

```diff
@@ -343,3 +343,101 @@ def enroll(uuid, organization, from_date=None, to_date=None):
+
+
+@django.db.transaction.atomic
+def withdraw(uuid, organization, from_date=None, to_date=None):
+    """Withdraw a unique identity from an organization.
+
+    This function withdraws a unique identity identified by `uuid`
+    from the given `organization` during the given period of time.
+
+    For example, if the unique identity `A` was enrolled from `2010-01-01`
+    to `2018-01-01` to the organization `Example`, the result of withdrawing
+    that identity from `2014-01-01` to `2016-01-01` will be two enrollments
+    for that identity: one for the period 2010-2014 and another one for
+    the period 2016-2018. If the period of withdrawing encloses minimum
+    and maximum dates, all the enrollments will be removed.
+
+    Both `uuid` and `organization` must exists before being deleted.
+    Moreover, an enrollment during the given period must exist.
+    Otherwise, it will raise a `NotFoundError` exception.
+
+    The period of the enrollment can be given with the parameters
+    `from_date` and `to_date`, where `from_date <= to_date`. Default
+    values for these dates are `1900-01-01` and `2100-01-01`.
+
+    The unique identity object with updated enrollment data is returned
+    as the result of calling this function.
+
+    :param uuid: unique identifier
+    :param organization: name of the organization
+    :param from_date: date when the enrollment starts
+    :param to_date: date when the enrollment ends
+
+    :returns: a unique identity with enrollment data updated
+
+    :raises NotFoundError: when either `uuid` or `organization` are not
+        found in the registry or when the identity is not enrolled
+        in that organization for the given period
+    :raises InvalidValeError: raised in three cases, when either identity or
+        organization are `None` or empty strings; when "from_date" < 1900-01-01 or
+        "to_date" > 2100-01-01; when "from_date > to_date"
+    """
+    if uuid is None:
+        raise InvalidValueError(msg="uuid cannot be None")
+    if uuid == '':
+        raise InvalidValueError(msg="uuid cannot be an empty string")
+    if organization is None:
+        raise InvalidValueError(msg="organization cannot be None")
+    if organization == '':
+        raise InvalidValueError(msg="organization cannot be an empty string")
+
+    from_date = datetime_to_utc(from_date) if from_date else MIN_PERIOD_DATE
+    to_date = datetime_to_utc(to_date) if to_date else MAX_PERIOD_DATE
+
+    if from_date < MIN_PERIOD_DATE or from_date > MAX_PERIOD_DATE:
+        raise InvalidValueError(msg="'from_date' date {} is out of bounds".format(from_date))
+    if to_date < MIN_PERIOD_DATE or to_date > MAX_PERIOD_DATE:
+        raise InvalidValueError(msg="'to_date' date {} is out of bounds".format(to_date))
+    if from_date > to_date:
+        msg = "'from_date' date {} cannot be greater than {}".format(from_date, to_date)
+        raise InvalidValueError(msg=msg)
+
+    # Find and check entities
```

Comment:
possible typo -> entities or identities?

Reply:
It's entities because the code looks for a unique identity and an organization.